corehq/pillows/tasks.py

```diff
@@ -30,3 +37,7 @@ def resave_es_forms_with_unknown_user_type(user_id):
     for domain, form_id in domain_form_id_list:
         form = FormAccessors(domain).get_form(form_id)
         resave_form(domain, form)
+
+
+def _get_unknown_user_type_user_ids():
+    return FormES().user_type('unknown').user_aggregation().run().aggregations.user.keys
```

Comment:
Is this only ever expected to return a few results? Is there no way to do a count query in ES?

Reply:
It initially receives 16K results on prod, and in steady state it currently returns 13.8K. This line is quite quick FWIW, but since that 13.8K can grow unbounded ("unknown" is currently the officially correct designation for any forms submitted by cases, which is an SMS-only workflow, though that in itself seems wrong), it does make me a little uneasy.

I just played around with it and I could get it to give me a good approximation with
```
FormES().user_type('unknown').aggregation(CardinalityAggregation('users_count', 'form.meta.userID')).run().aggregations.users_count.value
```
which uses an approximation algorithm, and compared to the real number it is good to 3 digits and only off by 29. There's a `precision_threshold` we can set to get it even closer, but I think for what we're using it for it doesn't matter much.

The only case when I think it could potentially be a problem is if the periodic queue is backed up for a long time and it runs a ton of these at the same time when it recovers. I'll use quickcache to make sure the function is only ever run no more frequently than 9 minutes apart.

Reply:
https://github.com/dimagi/commcare-hq/pull/26188